pipeline/pipeline_test.py

```diff
@@ -2,36 +2,78 @@
 
 from .pipeline import Pipeline
 from .task import AbstractTask
+from checkpoint_service import *
 
+TEST_CHECKPOINT_FILE = 'pipeline/test_data/pipeline_steps.log'
 
 class AppendTask(AbstractTask):
-    result = []
-
-    def __init__(self, name, number: int):
+    def __init__(self, name, number: int, result):
         super().__init__(name)
         self.number = number
+        self._result = result
 
     def run(self):
-        AppendTask.result.append(self.number)
-
+        self._result.append(self.number)
 
 class PipelineTest(unittest.TestCase):
+
+    def setUp(self):
+        if os.path.exists(TEST_CHECKPOINT_FILE):
+            os.remove(TEST_CHECKPOINT_FILE)
+
     def test_run(self):
-        task_1 = AppendTask("task1", 1)
-        task_2 = AppendTask("task2", 2)
-        task_3 = AppendTask("task3", 3)
-        task_4 = AppendTask("task4", 4)
+        result = []
+        task_1 = AppendTask("task1", 1, result)
+        task_2 = AppendTask("task2", 2, result)
+        task_3 = AppendTask("task3", 3, result)
+        task_4 = AppendTask("task4", 4, result)
 
-        pipeline = Pipeline('test_working_dir')
+        test_pipeline_steps_key_set = CheckpointKeySet(False, TEST_CHECKPOINT_FILE)
+        pipeline = Pipeline('test_data', test_pipeline_steps_key_set)
         node_1 = pipeline.add_task(task_1)
         node_2 = pipeline.add_task(task_2, [node_1])
         node_3 = pipeline.add_task(task_3, [node_1])
         pipeline.add_task(task_4, [node_2, node_3])
 
         pipeline.run()
 
-        self.assertEqual(AppendTask.result, [1, 2, 3, 4])
+        self.assertEqual(result, [1, 2, 3, 4])
+
+    def test_run_with_checkpoint(self):
+        result = []
+        task_1 = AppendTask("task1", 1, result)
```

Comment:
Thanks for the testing. Could you help extract common logic of pipeline creation into a function?